src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -685,6 +686,49 @@ void PolicyHandler::LinkAppsToDevice() {
   }
 }
 
+uint32_t PolicyHandler::GetApplicationIdForPolicyUpdate(
+    Applications app_list) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (app_list.empty()) {
+    return 0;
+  }
+
+  Applications::const_iterator it_app = app_list.begin();
+
+  const ApplicationSharedPtr application_ptr = *it_app;
+  const uint32_t app_id = application_ptr->app_id();
+
+  if (!application_ptr->IsRegistered()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Application " << app_id << " is not marked as registered.");
+    app_list.erase(it_app);
+    return GetApplicationIdForPolicyUpdate(app_list);
```

Comment:
@dev-gh I thin we can just loop over container elements even **without erasing**. We just need to find first app which suits our needs. If found - exit from the loop and proceed it, if not - next loop iteration.